src/idpyoidc/server/session/manager.py

```diff
@@ -95,7 +95,7 @@ def __init__(
 
         session_params = self.conf.get("session_params") or {}
         _crypt_config = get_crypt_config(session_params)
```

Comment:
it seems that `_crypt_config` is defined but not used later in the code.

If it is unused and it doesn't do any change in the current session, I propose its removal

Reply:
True, let's see if @rohe agrees with this change.

Reply:
The call to get_crypt_config() is there to provide a resonable default if parts of the configuration is missing.
You're removing that, why ?

Reply:
Where the configuration applies default stuffs, implicitly, I suggest to print an info log message to make the admins of the platform aware of that

there're a lot of configuration parameters, and I think that making the internal beahviour of idpy-oidc trnasparent and explicit in the logs will improve the usage experience of the solution

Reply:
We have three classes (we have more, but at least I will focus on those three):
1. `GrantManager`
2. `SessionManager`
3. `Database`

- `SessionManager` is a subclass of `GrantManager`.
- `GrantManager` is a subclass of `Database`.

In this part of the code we are initializing the `SessionManager`:
- To initialize the `SessionManager` we need to pass in a `handler`. Optionally, we may pass in a `conf` (and others which I will ignore for now). The `SessionManager` will call `super` that initializes its superclass, namely the `GrantManager`.
- The same holds for initializing the `GrantManager`. To initialize the `GrantManager` we need to pass in a `handler`. Optionally, we may pass in a `conf` (and others which I will ignore for now). The `GrantManager` will call `super` that initializes its superclass, namely the `Database`.
- To initialize the `Database` we need to pass in an optional `_crypt_config` and optionally `kwargs`.

Since `conf` is optional, we check whether it has a proper value, otherwise we default it. Because `conf` is an optional param, we need to do this check. But, this check happens on both `SessionManager` and `GrantManager` classes.

To present what happens in a more code-like fashion:

```py
init_session_manager(handler: required, conf_SM: optional):
  # as it is on SessionManager __init__
  # - check value of the conf param
  # - init session_params
  # - call get_crypt_config on session_params
  self.conf = conf_SM or {"session_params": {"encrypter": default_crypt_config()}}
  session_params = self.conf.get("session_params") or {}
  _crypt_config = get_crypt_config(session_params)            # first call

  # unroll the super(SessionManager, self).__init__(handler, _crypt_config) call
  # we are now initializing the GrantManager
  # as it is on GrantManager.__init__
  # - check value of the conf param
  # - init session_params
  # - call get_crypt_config on session_params

  # notice that the conf param is _crypt_config
  self.conf = _crypt_config or {                              # _crypt_config will always have a value prepared by the SessionManager
                                                              # so, self.conf will become _crypt_config
    "session_params": {
      "encrypter": default_crypt_config(),
      "node_type": ["client", "grant"],
      "node_info_class": {"client": ClientSessionInfo, "grant": Grant}
    }
  }

  session_params_GM = self.conf.get("session_params") or {}   # the session_params as prepared by the SessionManager
  _crypt_config_GM = get_crypt_config(session_params_GM)      # second call

  # unroll the super(GrantManager, self).__init__(_crypt_config, **self.conf) call
  ...
  if crypt_config_GM is None:
    crypt_config_GM = default_crypt_config()
```

Some things to note:
- checking the value of `conf` is repeated, but this is not needed
- the `SessionManager` cannot pass `kwargs` to the `Database`
- `default_crypt_config()` is called multiple times

---

> The call to get_crypt_config() is there to provide a resonable default if parts of the configuration is missing.
> You're removing that, why ?

`get_crypt_config()` is called on the superclass, that is on `GrantManager`. Removing it from the `SessionManager` allows the parent class to handle the call to `get_crypt_config`. It also allows any other kwargs set on `conf` that was passed to `SessionManager`, to reach the `Database` class.